symphony-bdk-core/src/main/java/com/symphony/bdk/core/activity/ActivityRegistry.java

```diff
@@ -44,31 +49,29 @@ public void register(final AbstractActivity<?, ?> activity) {
     this.activityList.add(activity);
   }
 
-  public List<AbstractActivity<? ,?>> getActivityList() {
+  public List<AbstractActivity<?, ?>> getActivityList() {
     return new ArrayList<>(activityList);
   }
 
   private void preProcessActivity(AbstractActivity<?, ?> activity) {
 
+    Optional<AbstractActivity<?, ?>> act = this.activityList.stream()
+        .filter(a -> a.getInfo().equals(activity.getInfo()))
+        .findFirst();
+
+    act.ifPresent(abstractActivity -> {
+      abstractActivity.bindToRealTimeEventsSource(this.datafeedLoop::unsubscribe);
+      this.activityList.remove(abstractActivity);
```

Comment:
let's add a debug log here to indicate we are replacing an existing activity this could help bot's developer to understand what is going on

Reply:
Will be added in next commit